sklearn/neighbors/_classification.py

```diff
@@ -277,7 +299,35 @@ def predict_proba(self, X):
             The class probabilities of the input samples. Classes are ordered
             by lexicographic order.
         """
+        check_is_fitted(self, "_fit_method")
         if self.weights == "uniform":
+            # TODO: systematize this mapping of metric for
+            # PairwiseDistancesReductions.
+            metric = _adjusted_metric(self.metric, self.p)
+            if self.metric == "precomputed":
+                X = _check_precomputed(X)
+
+            if (
+                self._fit_method == "brute"
+                and ArgKminClassMode.is_usable_for(X, self._fit_X, metric)
+                # TODO: Implement efficient multi-output solution
+                and not self.outputs_2d_
+            ):
+                metric_kwargs = self.metric_params or {}
+                if metric == "minkowski":
+                    metric_kwargs["p"] = self.p
```

Comment:
I think those 3 lines to prepare `metric_kwargs` would also deserve to be factored into the `_adjusted_metric` utility helper has well, no?